public/app/plugins/datasource/loki/querybuilder/state.ts

```diff
@@ -80,3 +80,28 @@ export function useRawQuery(): [boolean, (val: boolean) => void] {
 
   return [rawQuery, setter];
 }
+
+const queryEditorExplainLocalStorageKey = 'LokiQueryEditorExplainDefault';
+
+function getExplainVisibility(): boolean {
+  const val = store.get(queryEditorExplainLocalStorageKey);
+  return val === undefined ? true : Boolean(parseInt(val, 10));
```

Comment:
Do we want to show explain mode by default?

Reply:
Good question, maybe we can switch it to false. Yeah, I think default to false is fine.

Reply:
Do you want to do it as a part of moving hook to to shared file?

Reply:
Yeah I will do it in the hook ;)